src/charm.py

```diff
@@ -871,6 +896,19 @@ def _tsdb_versions_migration_dates(self) -> List[Dict[str, str]]:
         ret.append({"version": "v13", "date": tomorrow.strftime(date_format)})
         return ret
 
+    def _update_datasource_exchange(self) -> None:
+        """Update the grafana-datasource-exchange relations."""
+        if not self.unit.is_leader():
+            return
+
+        grafana_uids_to_units_to_uids = self.grafana_source_provider.get_source_uids()
+        raw_datasources: List[DatasourceDict] = []
+
+        for grafana_uid, ds_uids in grafana_uids_to_units_to_uids.items():
+            for _unit_name, ds_uid in ds_uids.items():
```

Comment:
I'm not saying we should copy-pasta the massive comment in the tempo code, but this is going to be confusing for everyone who doesn't know what's going on. Let's add a link to this post which we'll fill in in the coming days https://discourse.charmhub.io/t/tempo-ha-docs-correlating-traces-metrics-logs/16116

Reply:
added a comment with the ref